### DISCUSSION TITLE
Refactoring: Create event hub for system wide messaging

### DISCUSSION THREAD

**Discussion by Dangles91** [jellyfin/jellyfin-meta#52] [General]
I've been looking through the codebase and noticed there are a lot of direct dependencies just for subscribing to change events.

As the code base is difficult to work with due to wide spread dependencies, I think creating an event hub to handle system events would be a good first step in reducing the complexity.

I haven't worked out the exact design, but having some sort of EventAggregator or EventHub class with separate IEvent or IMessage interfaces for publishing and subscribing to events might be a good option. 

Event handlers (IEventHandler<IEvent>) could also be registered in the IOC Container to reduce the need for each handler to subscribe itself on init.

This design would not use the standard C# EventHandlers which may be undesirable for some developers, hence why I thought a discussion would be appropriate.

**Comment by crobibero** [comment]
We are already in the process of migrating to a Mediator pattern. 

https://github.com/jellyfin/jellyfin/pull/9305

**Comment by Dangles91** [comment]
That's excellent news. It looks like it's been stagnant for a while, but if it gets approval I'd be happy to assist with the migration.
